experiment/simulation/js/main.js

```diff
@@ -1 +1,318 @@
+      });
+
+    document.getElementById("instruction").innerHTML =
+      "Click on Observe button to observe what is happening inside the NMR Spectrometer and choose video speed according to your own liking.";
+
+    document.getElementById("observation").innerHTML =
+      "Click on Observe button to observe what is happening inside the NMR Spectrometer and choose video speed according to your own liking.";
+    overallIteration++;
+    restartAnimation = false;
+  }
+}
+
+let fillTube = async () => {
+  let path = document.getElementById("tube-path");
+  let finalPosition = 1;
+  let curPosition = 0;
+  while (true) {
+    if (curPosition > finalPosition) break;
+    curPosition += 0.01;
+    path.setAttribute("offset", curPosition);
+    await new Promise((resolve) => setTimeout(resolve, 0.5));
+  }
+};
+
+let fillPipette = async (color) => {
+  const line = document.getElementById("half-grad2");
+  const yFinalPosition = 0;
+  document.getElementById("line").style.stopColor = color;
+  let yPos = 100;
+  const interval = window.setInterval(() => {
+    if (yPos < yFinalPosition) {
+      line.setAttribute("y1", "0.1%");
+      return window.clearInterval(interval);
+    }
+    yPos -= 0.6;
+    line.setAttribute("y1", `${yPos}%`);
+  }, 1);
+};
+
+async function pourBenzene() {
+  if (overallIteration === 1) {
+    changeMessage();
+
+    let image = document.getElementById("pipette");
+    image.setAttribute("opacity", "1");
+    image.style.pointerEvents = "none";
+    let a1 = anime.timeline({
+      targets: "#pipette",
+      duration: 800,
+      easing: "linear",
+    });
+    let startX = "-490%";
+    let startY = "-100%";
+
+    let endX = "-430%";
+    let endY = "250%";
+
+    screenWidth();
+
+    a1.add({
+      duration: 0,
+      translateY: startY,
+      translateX: startX,
+    });
+    fillPipette(yellow);
+    await new Promise((r) => setTimeout(r, 1000));
+    a1.add({
+      duration: 500,
+      translateX: endX,
+    })
+      .add({
+        duration: 900,
+        translateY: endY,
+        update: function (anim) {
+          fillTube();
+        },
+      })
+      .add({
+        opacity: 0,
+      });
+
+    document.getElementById("tube").setAttribute("onclick", "moveTube()");
+    overallIteration++;
+    document.getElementById("benzene").style.cursor = "default";
+    document.getElementById("tube").style.cursor = "pointer";
+
+    if (restartAnimation) {
+      a1.restart();
+    }
+  }
+}
+
+let setupMessages = [
+  "Click on the Benzene option in the Apparatus Menu to introduce it into the workspace.",
+  "Click on the NMR Tube option in the Apparatus Menu to introduce it into the workspace.",
+  "Click on the NMR Spectrometer option in the Apparatus Menu to introduce it into the workspace.",
+];
+
+let setup = 0;
+
+function setupMessage() {
+  document.getElementById("instruction").innerHTML = setupMessages[setup];
+  document.getElementById("observation").innerHTML = setupMessages[setup];
```

Comment:
Please rethink this approach of using 2 separate html elements for one semantic element. Why cannot you just manage one element through CSS for different screen sizes?
This has the potential of opening up ugly bugs where any enhancements or fixes may work on one screen size and not the other.
If you still think you have to keep it, please document extensively so the maintainers have less chances of making a mistake.